src/gobupload/relate/__init__.py

```diff
@@ -12,52 +12,72 @@
 from gobcore.logging.logger import logger
 from gobcore.message_broker.config import CONNECTION_PARAMS, WORKFLOW_EXCHANGE, WORKFLOW_REQUEST_KEY
 from gobcore.message_broker.message_broker import Connection as MessageBrokerConnection
-from gobcore.model import GOBModel
 from gobcore.model.relations import get_relation_name
 from gobcore.sources import GOBSources
 from gobcore.typesystem import fully_qualified_type_name
 from gobcore.typesystem.gob_types import VeryManyReference
 
-from gobupload.relate.update import Relater
-from gobupload.relate.publish import publish_result
+from gobupload import gob_model
 from gobupload.storage.handler import GOBStorageHandler
 from gobupload.storage.materialized_views import MaterializedViews
 from gobupload.storage.relate import check_relations, check_very_many_relations, \
     check_relation_conflicts
 
+from gobupload.relate.update import Relater
+from gobupload.relate.publish import publish_result
+
 CATALOG_KEY = 'original_catalogue'
 COLLECTION_KEY = 'original_collection'
 ATTRIBUTE_KEY = 'original_attribute'
 RELATE_VERSION = '0.1'
 
 
 def check_relation(msg):
-    """
-    Check for any dangling relations
+    """Check for any dangling relations.
 
     :param msg:
     :return:
     """
-    header = msg.get('header', {})
-    catalog_name = header.get('original_catalogue')
-    collection_name = header.get('original_collection')
-    attribute_name = header.get('original_attribute')
-
-    model = GOBModel()
+    # Similar to _check_message()
+    try:
+        header = msg['header']
+    except KeyError as exc:
+        error_msg = "Invalid message: header key is missing"
+        _log_exception(error_msg, exc)
+        raise GOBException(error_msg) from exc
+
+    for key in [CATALOG_KEY, COLLECTION_KEY, ATTRIBUTE_KEY]:
+        try:
+            header[key]
+        except KeyError as exc:
+            error_msg = f"Missing required {key} attribute in message header"
+            _log_exception(error_msg, exc)
+            raise GOBException(error_msg) from exc
 
-    logger.info("Relate check started")
+    try:
+        catalog_name = header[CATALOG_KEY]
+        catalog = gob_model[catalog_name]
+    except KeyError as exc:
+        raise GOBException(f"Invalid catalog '{catalog_name}'") from exc
 
-    collection = model.get_collection(catalog_name, collection_name)
-    assert collection is not None, f"Invalid catalog/collection combination {catalog_name}/{collection_name}"
+    try:
+        collection_name = header[COLLECTION_KEY]
+        collection = catalog['collections'][collection_name]
+    except KeyError as exc:
+        raise GOBException(
+            f"Invalid collection '{collection_name}' for catalog {catalog_name}"
+        ) from exc
```

Comment:
I would extract this to a new function "verify_header" or something. Makes check_relation a whole lot shorter and more readable.

Even better (and easy to accomplish) would be to transform the header to a pydantic Header object: pydantic can do all the validation. You'll be sure that you have a validated Header object, plus you can use stuff like `header.attribute` below. Can also do a partial refactoring; when you don't want to refactor all the downstream code, you can still transform the pydantic object back to a dict in the return statement for example. But at least the validation is made much more simpler.